2024-05-30 00:08:24,374 ERROR: Exception on /user/Lauren [GET] [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\routes.py", line 90, in user
    return render_template('user.html', user=user, posts=posts, form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\templates\user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\templates\base.html", line 32, in top-level template code
    {% block content %} {% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\templates\user.html", line 36, in block 'content'
    {% include '_post.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\templates\_post.html", line 5, in template
    <a href="{{ url_for('user' username=post.author.username)}}">
    ^^^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.TemplateSyntaxError: expected token ',', got 'username'
2024-05-30 00:08:51,133 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:09:15,010 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:09:21,052 ERROR: Exception on /unfollow/Lauren [POST] [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\routes.py", line 140, in unfollow
    return redirect(url_for('user', username==username))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: url_for() takes 1 positional argument but 2 were given
2024-05-30 00:11:27,927 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:21:04,911 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:22:03,304 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:35:55,616 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:42:42,961 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:42:45,333 ERROR: Exception on /user/Joe_Mumna69420 [GET] [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py:838]
Traceback (most recent call last):
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\routes.py", line 89, in user
    posts = Post(body=form.post.data, author=current_user)
                      ^^^^
UnboundLocalError: cannot access local variable 'form' where it is not associated with a value
2024-05-30 00:43:02,778 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:53:42,185 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 00:57:56,092 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-05-30 01:06:50,973 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:47]
2024-06-01 13:27:48,072 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:49]
2024-06-01 13:58:06,446 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:49]
2024-06-01 13:58:59,817 INFO: Microblog Startup [in C:\Users\hazab\Desktop\VSCode Projects\FlaskTutorial\app\__init__.py:49]
